starlite/cache/base.py

```diff
@@ -1,51 +1,152 @@
-from typing import Any, Awaitable, overload
+from asyncio import Lock
+from typing import TYPE_CHECKING, Any, Optional, overload
 
 from typing_extensions import Protocol, runtime_checkable
 
+from starlite.utils import is_async_callable
+
+if TYPE_CHECKING:
+    from starlite.connection import Request
+    from starlite.types import CacheKeyBuilder
+
 
 @runtime_checkable
 class CacheBackendProtocol(Protocol):  # pragma: no cover
     @overload  # type: ignore[misc]
     def get(self, key: str) -> Any:
         ...
 
-    async def get(self, key: str) -> Awaitable[Any]:
-        """Retrieve value from cache corresponding to the given key.
+    async def get(self, key: str) -> Any:
+        """Retrieves a value from cache corresponding to the given key.
 
         Args:
-            key (str): name of cached value.
+            key: name of cached value.
 
         Returns:
-            Cached value.
+            Cached value if existing else `None`.
         """
 
     @overload  # type: ignore[misc]
     def set(self, key: str, value: Any, expiration: int) -> Any:
         ...
 
-    async def set(self, key: str, value: Any, expiration: int) -> Awaitable[Any]:
-        """Set a value in cache for a given key with a given expiration in
-        seconds.
+    async def set(self, key: str, value: Any, expiration: int) -> Any:
```

Comment:
Should this be `Awaitable[Any]`?

Reply:
It creates other issues. This way mypyc infers it correctly. Pylint doesn't, but it's less of an issue.